numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2674,16 +2654,30 @@ finish:
 #endif
     Py_DECREF(robj);
     return obj;
-#endif
 }
 #undef _@TYPE@_IS_@TYPE@
 
-#undef _NPY_UNUSED_TYPE
-#undef _NPY_UNUSED_KWDS
 /**end repeat**/
 
 #undef _WORK
 
+static PyObject *
+object_arrtype_new(PyTypeObject *NPY_UNUSED(type), PyObject *args, PyObject *kwds)
+{
+    PyObject *obj = Py_None;
+    char *kwnames[] = {"", NULL};  /* positional-only */
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|O:object_", kwnames, &obj)) {
```

Comment:
The old code optimized things more like `*obj = NULL` and then `if (obj == NULL) Py_RETURN_NONE;` that is probably not necessary, but seems also OK?

Reply:
It didn't seem particularly worthy of optimization to me, but I suppose someone might be calling `some_dtype.type()` in generic code to default-construct something. It's unlikely they're calling it more than a couple times though, as `zeros` would be more appropriate for that.

Reply:
No, I guess I agree, these are scalars. Creating a scalar with not obviously well defined data does indeed seem like a useless pattern.